### PR TITLE
[WIP] [Sorbet] Experiment

### PR DESCRIPTION
**Pull request by alloy**
https://sorbet.org/docs/quickref

### CODE REVIEW

File: sorbet/rbi/sorbet-typed/lib/actionpack/all/actionpack.rbi [hunk 1]
```diff
@@ -11,3 +11,17 @@ class ActionDispatch::Routing::RouteSet
...
+class ActionDispatch::Request
+  # sig {returns(ActionDispatch::Http::Headers)}
+  sig {returns(Hash)}
+  def headers; end
+end
+
+class ActionController::Base
+  sig {returns(ActionDispatch::Request)}
+  def request; end
+end
```
Comment (alloy) [inline sorbet/rbi/sorbet-typed/lib/actionpack/all/actionpack.rbi]: If you comment out these additions, you’ll see the following error show up:

```
$ bundle exec srb tc
app/controllers/api/graphql_controller.rb:11: Method request does not exist on Api::GraphqlController https://srb.help/7003
    11 |        user_agent: request.headers['User-Agent'],
                            ^^^^^^^
    https://github.com/sorbet/sorbet/tree/a2360f999e2e5ce19664b43932e7e9c216f71cfd/rbi/core/kernel.rbi#L758: Did you mean: Kernel#require?
     758 |  def require(path); end
            ^^^^^^^^^^^^^^^^^

app/controllers/api/graphql_controller.rb:12: Method request does not exist on Api::GraphqlController https://srb.help/7003
    12 |        user_ip: request.headers['x-forwarded-for']
                         ^^^^^^^
    https://github.com/sorbet/sorbet/tree/a2360f999e2e5ce19664b43932e7e9c216f71cfd/rbi/core/kernel.rbi#L758: Did you mean: Kernel#require?
     758 |  def require(path); end
            ^^^^^^^^^^^^^^^^^
Errors: 2
```

**Comment by alloy** [comment]
Nothing much yet, just a bootstrapped version of sorbet in exchange, for your entertainment.

File: Gemfile [hunk 2]
```diff
@@ -51,3 +51,8 @@ end
 
 gem 'sorbet', :group => :development
 gem 'sorbet-runtime'
+
+# These are needed when running `srb init`
+# gem 'cancan'
+# gem 'devise'
+# gem 'pundit'
```
Comment (alloy) [inline Gemfile]: https://github.com/sorbet/sorbet/issues/925#issuecomment-504208566

**Comment by alloy** [comment]
The repo that will hold community contributed typings for gems is still pretty empty, so looks like any use we'd need to write typings for initially https://github.com/sorbet/sorbet-typed

**Comment by ansor4** [comment]
Going to close this PR given ruby 3.0 comes with rbs. If we need sorbet in the future we can always refer back to this PR.

**Pull request closed without merging**